Core/src/IfcGeometryConverter/SolidModelConverter.h

```diff
@@ -174,9 +182,69 @@ namespace OpenInfraPlatform
 
 					BLUE_LOG(warning) << "Geometry conversion for IfcSectionedSolidHorizontal not implemented.";
 					// TO DO: implement, check for formal propositions. 
+                 
 
-				 //endif sectioned_solid_horizontal
-			}
+				//check dimensions and correct attributes sizes
+                if(vec_cross_sections.size() != vec_cross_section_positions.size())
+				{
+					std::cout << "CrossSections and CrossSectionsPositions are not equal in size " << std::endl;
+					return;
+				}
+				if ((vec_cross_sections.size() - vec_cross_section_positions.size()) > 1)
+				{
+					std::cout << "vec_cross_sections size decreased by one" << std::endl;
+					vec_cross_sections.pop_back();
+				}
+				else if ((vec_cross_section_positions.size() - vec_cross_sections.size()) > 1)
+				{
+					std::cout << "vec_cross_section_positions decreased by one" << std::endl;
+					vec_cross_section_positions.pop_back();
+				}

```

Comment:
This will never be executed (even in the previous version of the code), since the `return` or `throw` make the flow leave the function. These are not needed.

Reply:
ok, i changed the return for a oip::InconsistentModellingException. And if those two cases are not needed i will erase them as well in the code, i just thought they might be usefull.

Reply:
No need for them, erasing is just.

Reply:
resolved in commit ccd8948